benchmarking/evaluations/metrics/cosine_similarity.py

```diff
@@ -0,0 +1,87 @@
+import itertools
+import statistics
+from typing import List, Dict, Union, Tuple
+
+from numpy import dot
+from numpy.linalg import norm
+
+from api.models.student import Student
+from api.models.team import Team
+from api.models.team_set import TeamSet
+from benchmarking.evaluations.interfaces import TeamSetMetric
+
+
+class ClassAttributeSet:
+    def __init__(self, team_set: TeamSet):
+        self._data: Dict[Tuple[int, int], int] = {}
+        index = 0
+        for team in team_set.teams:
+            for student in team.students:
+                for key, values in student.attributes.items():
+                    for value in values:
+                        attribute = (key, value)
+                        if attribute not in self._data:
+                            self._data[attribute] = index
+                            index += 1
+
+    def index_of(self, attribute: Tuple[int, int]) -> int:
+        return self._data.get(attribute)
+
+    def size(self) -> int:
+        return len(self._data)
+
+    def get_student_vector(self, student: Student) -> List[int]:
+        v = [0] * self.size()
```

Comment:
because the way I generate the vector, all values should default to 0 (the student doesn't have the attrubute), then I update the indices where the student does have that attribute. So if I didn't do this, I would just have index out of bounds when trying to update attributes

Reply:
Ohhhh makes sense